enumerable_methods.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    my_each do |element|
+      result = yield(element)
+      next unless result
+
+      selected << element
+    end
+    selected
+  end
+
+  def my_all?
+    if block_given?
+      my_each { |element| return false if yield(element) == false }
+    else
+      my_each { |element| return false if element.nil? || element == false }
```

Comment:
This will give the same result since `element.nil? == (element == false) which is equal to false`. Hence we can return false when the `element` is falsy.

```suggestion
      my_each { |element| return false unless element }
```